File: C/algoritmo1.c
#include <stdio.h>

int main() {
    float entrada1, entrada2;

    printf("Digite o primeiro número: ");
    scanf("%f", &entrada1);

    printf("Digite o segundo número: ");
    scanf("%f", &entrada2);

    printf("Adição: %.2f + %.2f = %.2f\n", entrada1, entrada2, entrada1 + entrada2);
    printf("Subtração: %.2f - %.2f = %.2f\n", entrada1, entrada2, entrada1 - entrada2);
    printf("Multiplicação: %.2f * %.2f = %.2f\n", entrada1, entrada2, entrada1 * entrada2);

    // Verifica a divisão por zero antes de realizar a divisão
    if (entrada2 != 0) {
        printf("Divisão: %.2f / %.2f = %.2f\n", entrada1, entrada2, entrada1 / entrada2);
    }
    else {
        // Exibe uma mensagem de erro se a divisão por zero for tentada
        printf("Divisão: Erro. Não é possível dividir por zero.\n");
    }

    return 0;
}

File: Haskell/algoritmo1.hs
main :: IO ()
main = do
    putStrLn "Digite o primeiro número: "
    entrada1 <- readLn

    putStrLn "Digite o segundo número: "
    entrada2 <- readLn

    putStrLn $ "Adição: " ++ show (entrada1 + entrada2)
    putStrLn $ "Subtração: " ++ show (entrada1 - entrada2)
    putStrLn $ "Multiplicação: " ++ show (entrada1 * entrada2)

    if entrada2 /= 0
        then putStrLn $ "Divisão: " ++ show (entrada1 / entrada2)
        else putStrLn "Divisão: Erro. Não é possível dividir por zero."
